src/NServiceBus.Core/Audit/AuditToRoutingConnector.cs

```diff
@@ -15,38 +13,27 @@ public AuditToRoutingConnector(TimeSpan? timeToBeReceived)
             this.timeToBeReceived = timeToBeReceived;
         }
 
-        public override Task Invoke(IAuditContext context, Func<IRoutingContext, Task> stage)
+        public override async Task Invoke(IAuditContext context, Func<IRoutingContext, Task> stage)
         {
-            var message = context.Message;
-
-            if (context.Extensions.TryGet(out State state))
-            {
-                //transfer audit values to the headers of the message to audit
-                foreach (var kvp in state.AuditValues)
-                {
-                    message.Headers[kvp.Key] = kvp.Value;
-                }
-            }
-
             var dispatchProperties = new DispatchProperties();
 
             if (timeToBeReceived.HasValue)
             {
                 dispatchProperties.DiscardIfNotReceivedBefore = new DiscardIfNotReceivedBefore(timeToBeReceived.Value);
```

Comment:
Not sure if we want this abstracted as well, if so we need to either add it to the IAuditContext or just pass it to .GetRoutingData

Reply:
If we returned the routing context as in the code I showed you on friday it could be controlled inside the action

Reply:
this is now happening inside the action